core/src/test/java/org/mskcc/cbio/oncokb/util/AnnotationSearchUtilsTest.java

```diff
@@ -47,4 +49,42 @@ public void testSearchNonHgvsAnnotation() {
         // This is a variant search, the cancer type search result should not be available
         assertTrue(respSet.stream().filter(resp -> TypeaheadQueryType.CANCER_TYPE.equals(resp.getQueryType())).count() == 0);
     }
+
+    public void testAnnotationSearch() {
+        // test gene query
+        Set<AnnotationSearchResult> respSet = annotationSearch("BRAF");
+        // This is a gene search, there should be a gene search result
+        assertTrue(respSet.stream().filter(resp -> AnnotationSearchQueryType.GENE.equals(resp.getQueryType())).count() > 0);
+        // Since BRAF is mentioned in multiple alterations, variant search result should be available
+        assertTrue(respSet.stream().filter(resp -> AnnotationSearchQueryType.VARIANT.equals(resp.getQueryType())).count() > 0);
+        // BRAF is not part of any cancer type, the cancer type search result should not be available
+        assertTrue(respSet.stream().filter(resp -> AnnotationSearchQueryType.CANCER_TYPE.equals(resp.getQueryType())).count() == 0);
+
```

Comment:
Can we add 1 test here for `BCR-ABL1`. There should be gene and alteration results.

Reply:
The BCR-ABL1 should also have cancer type matches too